code/datums/components/shrink.dm

```diff
@@ -23,19 +24,23 @@
 			if(ishuman(C))
 				var/mob/living/carbon/human/H = C
 				H.physiology.damage_resistance -= 100//carbons take double damage while shrunk
+	else
+		parent_atom.set_density(FALSE) // this is handled by the UNDENSE trait on mobs
```

Comment:
You already call this when adding/removing the density traits, right? Why not just add/remove traits here? (I think lemon also asked is)

Reply:
The density traits only apply to mobs, this line is only called if (isliving(parent_atom)) comes out false.